### DISCUSSION TITLE
Best way to use graphql?

### DISCUSSION THREAD

**Discussion by krymancer** [solidjs/solid#1156] [Q&A]
So i'm using (solid-urql)[https://www.npmjs.com/package/solid-urql] but i have somethings that i cant do easly with it, like refetch queries. I thinking abour using urql and createResource, but i'm not sure if is a good ideia, because i dont know how to provide de same client to the entire app. I guess i cloud use a store, but dont know if is the best option.

**Comment by frederikhors** [comment]
@Krymancer did you find a way?

**Reply by krymancer** [reply]
I ended up using [solid-primitives/gralphql](https://www.npmjs.com/package/@solid-primitives/graphql)

You can create a graphqlClient, create a query and using signals i was able to invalidade previous state to fetch new data. Not quite the best solution, but worked for me at the time.

It's been an while the last time that i use graphql so must be a better way to do this now, like resources of some other new feature of solid

**Reply by frederikhors** [reply]
Yeah, what I would like is a normalized cache like urql does. Thank you anyway.
